### DISCUSSION TITLE
RmlUI 的鼠标悬停事件

### DISCUSSION THREAD

**Discussion by cloudwu** [ejoy/ant#194] [Ideas]
之前因为一直在手机上做 UI ，所以忽略了鼠标悬停事件。最近因为在 Windows 上开发，这个还是很有必要的。

https://github.com/ejoy/ant/commit/3d008d0bd5c4c13bc6c89886f937c24570b782a7

我简单给 RmlUI 加了一个 hover 事件，目前在鼠标移动到控件上，以及移出控件都会触发 `data-event-hover`  ，区别在于 ev.state 是 enter 还是 leave 。这个事件的用法和 `data-event-click` 类似 。

在控件上移动，但没有移出去，是没有额外事件的。

@actboy168 看看？

现在 tap 会触发默认行为，使用 active 的子 style 。我想 hover 时候也需要有个类似的默认行为？我不熟悉 web 开发，通常 html 是怎样支持鼠标悬停时采用不同 style 的？

**Comment by cloudwu** [comment]
现在支持定义 `buttom:hover` 这样的 style ，供鼠标悬停的时候使用。和原来的 button:active 类似 ： https://github.com/ejoy/ant/blob/master/test/rmlui/pkg/ant.test.rmlui/start.html#L26

**Comment by actboy168** [comment]
曾经有 hover，后来刻意删掉了。也许可以从历史记录里把它找回来
